options/rtdl/generic/main.cpp

```diff
@@ -527,7 +528,7 @@ int __dlapi_iterate_phdr(int (*callback)(struct dl_phdr_info *, size_t, void*),
 	for (auto object : globalScope->_objects) {
 		struct dl_phdr_info info;
 		info.dlpi_addr = object->baseAddress;
-		info.dlpi_name = object->name;
+		info.dlpi_name = object->name.data(); // TODO: this pointer will dangle if we modify object->name
```

Comment:
I don't think this is a concern for us; I can't think of a reason why the name we recorded of an already-loaded library would change.

Reply:
The way musl does it (and the way I implement it in #448) they do modify the recorded name if a DSO is loaded first by path (e.g `foo/bar.so`) and then by 'shortname' (`bar.so`), then the recorded name will be modified to `bar.so`. Another case where we'd need to be careful about `.data()` is if we ever implement dlclose and free the backing memory for the object's name. It's true that we don't implement either of these *yet* but it's probably prudent to keep the comment here.